flink-core/src/main/java/org/apache/flink/api/common/io/OutputFormatBase.java

```diff
@@ -62,7 +70,10 @@ protected OutputFormatBase(int maxConcurrentRequests, Duration maxConcurrentRequ
         this.maxConcurrentRequestsTimeout = maxConcurrentRequestsTimeout;
     }
 
-    /** Opens the format and initializes the flush system. */
+    /**
+     * Open the format and initializes the flush system. Implementers must call {@code
+     * super.open()}.
+     */
```

Comment:
You could enforce this by making open() final and adding another initialization hook for sub-classes that we call at the end of `open()`.

Reply:
thanks for the suggestion.

Reply:
:+1: